adapters/revcontent/revcontent.go

```diff
@@ -0,0 +1,124 @@
+	"net/http"
+)
+
+type RevcontentAdapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the Revcontent adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	bidder := &RevcontentAdapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (adapter *RevcontentAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var adapterRequests []*adapters.RequestData
+
+	adapterRequest, errs := adapter.makeRequest(request)
+	if errs == nil {
+		adapterRequests = append(adapterRequests, adapterRequest)
+	}
+
+	return adapterRequests, errs
+}
+
+func (adapter *RevcontentAdapter) makeRequest(request *openrtb.BidRequest) (*adapters.RequestData, []error) {
+	var errs []error
+
+	reqBody, err := json.Marshal(request)
+
+	if err != nil {
+		return nil, append(errs, err)
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+
+	return &adapters.RequestData{
+		Method:  "POST",
+		Uri:     adapter.endpoint,
+		Body:    reqBody,
+		Headers: headers,
+	}, errs
+}
+
+// MakeBids make the bids for the bid response.
+func (a *RevcontentAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Unexpected status code: %d.", response.StatusCode),
+		}}
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d.", response.StatusCode),
+		}}
+	}
+
+	var bidResp openrtb.BidResponse
+
+	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
+		return nil, []error{err}
+	}
+
+	count := getBidCount(bidResp)
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(count)
+
+	for _, sb := range bidResp.SeatBid {
+		for i := range sb.Bid {
+			var mediaType = getBidType(sb.Bid[i].ImpID, internalRequest.Imp)
+			bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+				Bid:     &sb.Bid[i],
+				BidType: mediaType,
+			})
+		}
+	}
+	return bidResponse, nil
+
+}
+
+func getBidCount(bidResponse openrtb.BidResponse) int {
+	c := 0
+	for _, sb := range bidResponse.SeatBid {
+		c = c + len(sb.Bid)
+	}
+	return c
+}
+
+func getBidType(impId string, imps []openrtb.Imp) openrtb_ext.BidType {
+	bidType := openrtb_ext.BidTypeBanner
+	for _, imp := range imps {
+		if imp.ID == impId {
+			if imp.Banner != nil {
+				break
+			}
+			if imp.Video != nil {
+				bidType = openrtb_ext.BidTypeVideo
+				break
+			}
+			if imp.Native != nil {
+				bidType = openrtb_ext.BidTypeNative
+				break
+			}
+
+		}
+	}
+	return bidType
```

Comment:
It would be ideal to use a response from your bidder to set the media type. Does your bidder include any ext values to use in that determination? If not, what would your bidder respond back with if the impression is for both banner and native?